### PR TITLE
chore(website): swizzle 404 page with editor-like squiggle

### LINKED ISSUE #7346: Website: Make a fun 404 page
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Our 404 page is very boring right now:

> **Page Not Found**
> We could not find what you were looking for.
>
> Please contact the owner of the site that linked you to the original URL and let them know their link is broken.

I've worked on a few fun 404s before:

* codecademy.com/404
* learningtypescript.com/404

Surely we can come up with a more fun one? 😄 

### Affected URL(s)

http://localhost:3000/any-404-really

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #7346
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Adds a 404 page somewhat like the one on learningtypescript.com/404.

Screenshot of a light mode 404 page saying '/wat' is not defined. The pathname has a red squiggly below it and '$npx eslint .' above it.

Screenshot of a dark mode 404 page saying '/blog/much-longer-title-split-across-lines' is not defined. The pathname has a red squiggly below it and '$npx eslint .' above it.

The squiggle asset is similar to the one I use on joshuakgoldberg.com.

### CODE REVIEW

File: packages/website/src/theme/NotFound/styles.module.css [hunk 1]
```diff
@@ -0,0 +1,28 @@
+.title {
+  font-family: monospace;
+  font-size: 3rem;
+  margin-bottom: 2rem;
+}
+
+.title::before {
+  content: '$ npx eslint .';
```
Comment (Josh-Cena) [inline packages/website/src/theme/NotFound/styles.module.css:8]: Why is this in CSS not in markup? 🤔

File: packages/website/src/theme/NotFound/styles.module.css [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+.word::after {
+  content: '';
+  display: block;
+  position: absolute;
+  top: 100%;
+  left: 0;
+  width: 100%;
+  height: 13px;
+  background: url('/img/squiggle.svg') repeat-x;
+  background-size: contain;
```
Comment (Josh-Cena) [inline packages/website/src/theme/NotFound/styles.module.css:26]: Nice CSS trick!

File: packages/website/src/theme/NotFound/index.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  return (
+    <>
+      <PageMetadata title="Page Not Found" />
+      <Layout>
+        <main className="container margin-vert--xl">
+          <div className="row">
+            <div className="col col--8 col--offset-2">
+              <h1 className={styles.title}>
+                <strong>
+                  {`'${location.pathname}'`.split('').map((letter, i) => (
```
Comment (Josh-Cena) [inline packages/website/src/theme/NotFound/index.tsx:20]: Tiny tiny nit: a string cannot possibly be undefined. Why don't we get rid of `''` and pretend we are in a language where `/` and `-` are valid identifier parts? Just an opinion 😄

File: packages/website/src/theme/NotFound/index.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import { useLocation } from '@docusaurus/router';
+import { PageMetadata } from '@docusaurus/theme-common';
+import Layout from '@theme/Layout';
+import React from 'react';
+
+import styles from './styles.module.css';
+
+export default function NotFound(): React.JSX.Element {
```
Comment (Josh-Cena) [inline packages/website/src/theme/NotFound/index.tsx:8]:
```suggestion
export default function NotFound(): JSX.Element {
```

Comment (bradzacher) [reply packages/website/src/theme/NotFound/index.tsx:8]: `React.` is actually the new way of doing things - the global `JSX` type is deprecated.

Comment (Josh-Cena) [reply packages/website/src/theme/NotFound/index.tsx:8]: Ah, didn't know that. This is so much longer though :(

File: packages/website/src/theme/NotFound/styles.module.css (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/website/src/theme/NotFound/styles.module.css:8]: No particular reason 😄 screen readers do announce them. I'll just move it to markup.

Comment (JoshuaKGoldberg) [reply packages/website/src/theme/NotFound/styles.module.css:8]: Oh, right, it's that Docusaurus adds default styles to `<code>` and `<pre>` blocks. Switched to a `<div>`.

File: packages/website/src/theme/NotFound/styles.module.css (hunk 2 above)
Comment (JoshuaKGoldberg) [reply packages/website/src/theme/NotFound/styles.module.css:26]: Thanks! I use it on joshuakgoldberg.com 😄

File: packages/website/src/theme/NotFound/index.tsx (hunk 3 above)
Comment (JoshuaKGoldberg) [reply packages/website/src/theme/NotFound/index.tsx:20]: Sorry, I want to go along with the happy vibe of this suggestion but I don't understand what change it's asking for 😅. `.split()` isn't allowed by TypeScript.

File: packages/website/src/theme/NotFound/index.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  const location = useLocation();
```
Comment (JoshuaKGoldberg) [inline packages/website/src/theme/NotFound/index.tsx:9]: Aha: https://github.com/facebook/docusaurus/issues/9170

File: packages/website/src/theme/NotFound/index.tsx (hunk 3 above)
Comment (Josh-Cena) [reply packages/website/src/theme/NotFound/index.tsx:21]:
```suggestion
                  {location.pathname.split('').map((letter, i) => (
```

Like this. Sorry I wasn't clear.

Comment (Josh-Cena) [reply packages/website/src/theme/NotFound/index.tsx:21]: Actually never mind, I misread. I thought in `'/foo' is not defined`, `'/foo'` is a string literal, but that's how ESLint's error message actually reports an invalid identifier.

**Review by Josh-Cena — APPROVED** [review]
Very cool!

**Pull request merged**